src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -785,8 +787,12 @@ abstract class UnCurry extends InfoTransform
       if (dd.symbol.isConstructor)
         reporter.error(dd.symbol.pos, "A constructor cannot be annotated with a `varargs` annotation.")
       else {
-        val hasRepeated = mexists(dd.symbol.paramss)(sym => definitions.isRepeatedParamType(sym.tpe))
-        if (!hasRepeated) reporter.error(dd.symbol.pos, "A method without repeated parameters cannot be annotated with the `varargs` annotation.")
+        val ok = cond(dd.symbol.paramss.filter(_.nonEmpty)) {
+          case initPs :+ lastPs =>
+            initPs.forall(!definitions.isVarArgsList(_)) && definitions.isVarArgsList(lastPs)
+        }
+        if (!ok)
+          reporter.error(dd.pos, "A method annotated with @varargs must have a single repeated parameter in its last parameter list.")
```

Comment:
The message is no longer in sync with the implementation.

Reply:
"... in its last non-empty parameter list" is( even more of) a mouthful, and the more I think about it the less I like that you can even do this.

Reply:
I can't say I care all that much, but like you said why forbid it when it works.

Reply:
The varargs annotation is just about the java forwarder, which doesn't see the final empty parameter list at all. In scala, you can have varargs in any parameter list.

I had once upon a time "must have a single repeated parameter as its final parameter" -- because that's what matters for the java forwarder.

Reply:
I think it's better to keep the simpler wording here.